casepro/backend/junebug.py

```diff
@@ -0,0 +1,522 @@
+from datetime import datetime
+from django.conf import settings
+from django.conf.urls import url
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+import json
+import requests
+import pytz
+
+from . import BaseBackend
+from ..contacts.models import Contact
+from ..msgs.models import Message
+from ..utils import uuid_to_int
+
+from dash.utils import is_dict_equal
+from dash.utils.sync import BaseSyncer, sync_local_to_changes
+
+from itertools import chain
+
+
+class IdentityStore(object):
```

Comment:
As I mentioned in dev list email, we would have added a URNs/addresses to the contact model in CasePro except that they're not needed in the UPartners case, which requires names and phone numbers to be hidden. Is that something that would be useful for you guys to add now, so that they can be kept in sync with this identity store and more easily surfaced in the UI at some point?